app/javascript/mastodon/features/emoji/__tests__/emoji-test.js

```diff
@@ -57,5 +57,10 @@ describe('emoji', () => {
     it('does an emoji whose filename is irregular', () => {
       expect(emojify('↙️')).toEqual('<img draggable="false" class="emojione" alt="↙️" title=":arrow_lower_left:" src="/emoji/2199.svg" />');
     });
+
+    it('avoid emojifying on invisible text', () => {
+      expect(emojify('<a href="http://example.com/test%F0%9F%98%84"><span class="invisible">http://</span><span class="ellipsis">example.com/te</span><span class="invisible">st😄</span></a>'))
+        .not.toEqual('<a href="http://example.com/test%F0%9F%98%84"><span class="invisible">http://</span><span class="ellipsis">example.com/te</span><span class="invisible">st<img draggable="false" class="emojione" alt="😄" title=":smile:" src="/emoji/1f604.svg" /></span></a>');
```

Comment:
this should probably be `.toEqual('<a href="http://example.com/test%F0%9F%98%84"><span class="invisible">http://</span><span class="ellipsis">example.com/te</span><span class="invisible">st😄</span></a>')` instead of `.not.toEqual()`, to be more resilient to changes in the emojify code (reordering the attributes and then regressing would still cause this test to pass)